experimental/trusted_shuffler/server/src/http.rs

```diff
@@ -59,7 +59,27 @@ impl RequestHandler for HttpRequestHandler {
 
         log::info!("New request to the backend: {:?}", request);
 
-        match send_with_request(request).await {
+        // If the backend_url is a https connection, we need to use the AlpnConnector for the
+        // client connecting to the backend.
+        let is_https_url: bool = request
+            .uri()
+            .scheme()
+            .expect("Could not get scheme from backend_url.")
+            == &http::uri::Scheme::HTTPS;
+
+        let response = if is_https_url {
+            Client::builder()
+                .http2_only(true)
+                .build(AlpnConnector::new())
+                .request(request)
+        } else {
+            Client::builder()
+                .http2_only(true)
+                .build_http()
+                .request(request)
+        };
```

Comment:
I think this would look cleaner if it was conditionally building the client first and then sending the request.

```rust
let mut client_builder = Client::builder().http2_only(true);

let client = if is_https_url { client_builder.build(AlpnConnector::new()) } else { client_builder.build_http() };

let response = client.request(request);
```

Reply:
I wish, yes, but because the `client_builder` is either `Client<AlpnConnector, {unknown}>` or `Client<HttpConnector, {unknown}>` and wasn't could not to convince the type checker it is fine. 

I also figured we'd want to revisit that anyway: It might be better to do that check and choice only when starting the Trusted Shuffler, not for every request.